aten/src/ATen/native/Distributions.cpp

```diff
@@ -132,7 +132,7 @@ Tensor& bernoulli_out(Tensor& result, const Tensor& self, Generator* gen) {
 }
 
 Tensor& bernoulli_tensor_cpu_(Tensor& self, const Tensor& p_, Generator* gen) {
-  AT_DISPATCH_ALL_TYPES(self.scalar_type(), "bernoulli_tensor_cpu_self_", [&] {
+  AT_DISPATCH_ALL_TYPES_AND(at::ScalarType::Bool, self.scalar_type(), "bernoulli_tensor_cpu_self_", [&] {
```

Comment:
Why is there a difference between CPU and CUDA `bernoulli_` implementations w.r.t. supporting half precision?

Reply:
I think a lot ops are not supported for CPU half due to inefficiency.